### ISSUE TITLE
Avoiding excessive calls to MPI_Comm_dup

### ISSUE THREAD

**Issue by poulson** [elemental/Elemental#89]
Currently ``DistMultiVec``, ``DistSparseMatrix``, ``DistGraph``, etc., all construct their own duplicate of the MPI communicator passed into their constructor. Given that this occurs many thousands (perhaps millions or billions?) of times within a typical call to one of Elemental's IPMs, it is likely worthwhile to create an anologue of the Grid& class meant for the sparse objects (i.e., something other than a flat ``mpi::Comm`` which can be expected to stay in scope and does not need to be duplicated).

**Comment by jeffhammond** [comment]
Could you wrap MPI's comm with a reference counter and make Elemental's comm dup a shallow copy, at least under some circumstances where you know that the pitfalls of this will be avoided?

**Comment by poulson** [comment]
Yes, though the current massive migration of distributed data structures will handle the problem via switching to passing around ``Grid`` references.

Your suggested approach certainly takes less code though!

I hope to have integrated DistMultiVec into DistMatrix within the next couple of days
